slides/aml-16-032118-clustering-and-mixture-models/index.html

```diff
@@ -0,0 +1,756 @@
+???
+
+---
+
+# Pros and Cons
+
+- Can restrict to input “topology” given by any
+graph, for example neighborhood graph.
+- Fast with sparse connectivity, otherwise O(log(n) n
+** 2 )
+- Some linkage criteria an lead to very imbalanced
+cluster sized (can be pro or con)
+- Hierarchical clustering gives more holistic view, can
+help with picking the number of clusters.
+
+???
+
+---
+class: center, middle
+
+## DBSCAN
+
+???
+
+---
+
+# Algorithm
+
+.left-column[
+.smaller[
+- Clusters are formed by “core samples”
+- Sample is “core sample” if more than min_samples is
+within epsilon - “dense region”
+- Start with a core sample
+- Recursively walk neighbors that are core-samples and
+add to cluster.
+- Also add samples within epsilon that are not core
+samples (but don’t recurse)
+- If can’t reach any more points, pick another core
+sample, start new cluster.
+- Remaining points are labeled outliers
+]
+]
+
+.right-column[
+.center[
+![:scale 80%](images/dbscan_algo.png)
+]
+]
+
+???
+
+---
+
+# Illustration
+
+.center[
+![:scale 70%](images/dbscan_illustration.png)
+]
+
+???
+
+---
+
+# Pros and Cons
+
+- Allows complex cluster shapes
+- Can detect outliers
+- Needs two parameters to adjust, eps is hard to
+pick (can be done based on number of clusters
+though).
+- Can learn arbitrary cluster shapes
+
+.center[
+![:scale 40%](images/dbscan_pro_con.png)
+]
+
+???
+
+---
+class: center, middle
+
+## (Gaussian) Mixture Models
+
+???
+
+---
+
+# Mixture Models
+
+- Generative model – find p(X).
+- Assume form of data generating process
+- Mixture model assumption:
+  - Data is mixture of small number of known distributions.
+  - Each mixture component distribution can be learned "simply"
+  - Each point comes from one particular component
+- We learn the component parameters and weights of
+components
+
+` $$ p(\mathbf{x}) = \sum_{j=1}^k \pi_k p_k(\mathbf{x} | \theta) $$ `
+
+???
+
+---
+
+# Gaussian Mixture Models
+
+- Each component is created by a Gaussian distribution.
+- There is a multinomial distribution over the components
+
+`$$ p(\mathbf{x}) = \sum_{j=1}^k \pi_k \mathcal{N}(\mathbf{x}, \mu_k, \Sigma_k) $$`
+
+- Non-convex optimization.
+- Alternately assign points to components and compute mean
+and variance (EM algorithm).
+- Initialized with K-means, random restarts
+
+???
+
+---
+
+`$$ p(\mathbf{x}) = \sum_{j=1}^k \pi_k \mathcal{N}(\mathbf{x}, \mu_k, \Sigma_k) $$`
+
+.center[
+![:scale 40%](images/gmm1.png)
+]
+
+.center[
+![:scale 40%](images/gmm2.png)
+]
+
+???
+
+---
+
+# Goals of Mixture Models
+
+- Create parametric density model.
+- Allows for testing how “likely” a new point is.
+- Clustering (each components is one cluster).
+- Feature Extraction
+
+.center[
+![:scale 80%](images/gmm2.png)
+]
+
+???
+
+---
+
+# Examples
+
+.wide-left-column[
+.smaller[
+```python
+from sklearn.mixture import GaussianMixture
+gmm = GaussianMixture(n_components=3)
+gmm.fit(X)
+print(gmm.means_)
+print(gmm.covariances_)
+```
+[[-2.286 -4.674] </br>
```

Comment:
can you format these in triple backticks so the alignment works out? Then you don't need the <br> I think.

Reply:
Cool. I have added the triple backticks instead of breaks.